clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/SimpleClusterTierClientEntity.java

```diff
@@ -69,18 +70,16 @@
   private final ReconnectMessageCodec reconnectMessageCodec = new ReconnectMessageCodec();
   private final Map<Class<? extends EhcacheEntityResponse>, List<ResponseListener<? extends EhcacheEntityResponse>>> responseListeners =
     new ConcurrentHashMap<>();
+  private final List<DisconnectionListener> disconnectionListeners = new ArrayList<>();
```

Comment:
No concurrency possible on adding listeners? Additions are done from the same thread using the listener? (otherwise, you have visibility issues here)

Reply:
No since these all are one time (at the time of creation only), which is under lock